predicates/query/query.go

```diff
@@ -0,0 +1,110 @@
+/*
+Package source implements a custom predicate to match routes
+based on the Query Params in URL
+
+It supports checking existence of query params and also checking whether
+query params value match to a given regular exp
+
+Examples:
+
+    // Checking existence of a query param
+    // matches http://example.org?bb=a&query=withvalue
+    example1: QueryParam("query") -> "http://example.org";
+
+    //Even a query param without a value
+    // matches http://example.org?bb=a&query=
+    example1: QueryParam("query") -> "http://example.org";
+
+    //matches with regexp
+    // matches http://example.org?bb=a&query=example
+    example1: QueryParam("query", "^example$") -> "http://example.org";
+
+    //matches with regexp and multiple values of query param
+    // matches http://example.org?bb=a&query=testing&query=example
+    example1: QueryParam("query", "^example$") -> "http://example.org";
+
+*/
+
+package query
+
+import (
+	"github.com/zalando/skipper/predicates"
+	"github.com/zalando/skipper/routing"
+	"net/http"
+	"regexp"
+)
+
+type MatchType int
+
+const (
+	Exists  MatchType = 1
+	Matches MatchType = 2
+)
+
+type predicate struct {
+	typ       MatchType
+	paramName string
+	valueExp  *regexp.Regexp
+}
+type spec struct{}
+
+const name = "QueryParam"
+
+func New() routing.PredicateSpec { return &spec{} }
+
+func (s *spec) Name() string {
+	return name
+}
+
+func (s *spec) Create(args []interface{}) (routing.Predicate, error) {
+	lengs := len(args)
```

Comment:
interesting var name `lengs`, if it's a signature habit, i like it. But typically len(slice) doesn't need to be put in a var, because it is a builtin syntax. nicer to just use len(args).